File: C++/1684. Count the Number of Consistent Strings.cpp
class Solution {
public:
    int countConsistentStrings(std::string allowed, std::vector<std::string>& words) {
    int consistentCount = 0;
    bool allowedChars[26] = {false};
    for (char ch : allowed) {
        allowedChars[ch - 'a'] = true;
    }
    for (const std::string& word : words) {
        bool isConsistent = true;
        for (char ch : word) {
            if (!allowedChars[ch - 'a']) {
                isConsistent = false;
                break;
            }
        }
        if (isConsistent) {
            consistentCount++;
        }
    }
    return consistentCount;
    }
};

File: JAVA/1684. Count the Number of Consistent Strings.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        int count=0,no=0;
        int finall=1;
        char arr[]=new char[allowed.length()];
        for(int i=0;i<allowed.length();i++){
            arr[i]=allowed.charAt(i);
        }
        for(int i=0;i<words.length;i++){
            String str=words[i];
            finall=1;
            for(int j=0;j<words[i].length();j++){
                char c=words[i].charAt(j);
                int found=0;
                for(int k=0;k<arr.length;k++){
                    if(c==arr[k]){
                        found=1;
                        break;
                    }
                }
                if(found==0){
                        no++;
                        finall=0;
                        break;
                    }
            }
            if(finall!=0){
                count++;
            }
        }
        return words.length-no;
    }
}
